File: javascript/solutions/plus-minus.js
function plusMinus(arr) {
  let length = arr.length;
  let pos = 0;
  let neg = 0;
  let zed = 0;
  
  for (let i = 0; i < arr.length; i++) {
      if (arr[i] == 0) {
          zed++;
      } if (arr[i] > 0) {
        pos++
      } else if (arr[i] < 0) {
        neg++;
      }
  }

  let ratPos = (pos / length).toFixed(6)
  let ratNeg = (neg / length).toFixed(6)
  let ratZed = (zed / length).toFixed(6)

  console.log(`${ratPos}\n${ratNeg}\n${ratZed}`);
}

const arr1 = [1, -2, 3, -4, 5]
const arr2 = [7, 8, 0 ]
const arr3 = [1, -1, 1, 0, 0, ]

plusMinus(arr1)
plusMinus(arr2)
plusMinus(arr3)

File: kotlin/solutions/plus-minus.kt
fun plusMinus(arr: List<Int>): Unit {
    var (pos, neg, zed) = listOf(0, 0, 0)

    for (num in arr) {
        if (num > 0) ++pos
        if (num < 0) ++neg
        if (num == 0) ++zed
    }

    val result = listOf(pos, neg, zed)

    for (count in result) {
        var fraction = count.toDouble() / arr.size
        var strFraction = "%.5f".format(fraction)
        println(strFraction)
    }
}

fun main() {
    plusMinus(listOf(1, 3, 0, -10, -1000, 5, 0))
}